Sources/InfluxDBSwift/Point.swift

```diff
@@ -163,6 +125,169 @@ extension InfluxDBClient {
+    /// Possible value types of Field
+    public enum FieldValue {
+        /// Support for Int8
+        init(_ value: Int8) {
+            self = .int(Int(value))
+        }
+        /// Support for Int16
+        init(_ value: Int16) {
+            self = .int(Int(value))
+        }
+        /// Support for Int32
+        init(_ value: Int32) {
+            self = .int(Int(value))
+        }
+        /// Support for Int64
+        init(_ value: Int64) {
+            self = .int(Int(value))
+        }
+        /// Support for UInt8
+        init(_ value: UInt8) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt16
+        init(_ value: UInt16) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt32
+        init(_ value: UInt32) {
+            self = .uint(UInt(value))
+        }
+        /// Support for UInt64
+        init(_ value: UInt64) {
+            self = .uint(UInt(value))
+        }
+        /// Support for Float
+        init(_ value: Float) {
+            self = .double(Double(value))
+        }
+
+        /// signed integer number
+        case int(Int)
+        /// unsigned integer number
+        case uint(UInt)
+        /// floating number
+        case double(Double)
+        /// true or false value
+        case boolean(Bool)
+        /// string value
+        case string(String)
+    }
+
+    /// Possible value types of Field
+    public enum TimestampValue: CustomStringConvertible {
+        // The number of ticks since the UNIX epoch. The value has to be specified with correct precision.
+        case interval(Int, InfluxDBClient.TimestampPrecision = InfluxDBClient.defaultTimestampPrecision)
+        // The date timestamp.
+        case date(Date, InfluxDBClient.TimestampPrecision = InfluxDBClient.defaultTimestampPrecision)
+
+        public var description: String {
+            switch self {
+            case let .interval(ticks, precision):
+                return "\(ticks) [\(precision)]"
+            case let .date(date, precision):
+                return "\(date) [\(precision)]"
+            }
+        }
+
+        public var precision: InfluxDBClient.TimestampPrecision {
+            switch self {
+            case let .interval(_, precision):
+                return precision
+            case let .date(_, precision):
+                return precision
+            }
+        }
+    }
+}
+
+extension InfluxDBClient.Point {
+    // swiftlint:disable large_tuple cyclomatic_complexity function_body_length
+
+    /// Create a new Point from Tuple.
+    ///
+    /// - Parameters:
+    ///   - tuple: the tuple with keys: `measurement`, `tags`, `fields` and `time`
+    ///   - precision: the data point precision
+    /// - Returns: created Point
+    public class func fromTuple(
+            _ tuple: (
+                    measurement: String,
+                    tags: [String?: String?]?,
+                    fields: [String?: Any?], time: Any?),
```

Comment:
These tuple elements should be strongly typed, using `FieldValue` and `TimestampValue`

Reply:
Now I am using strongly typed tuple:

```swift
public typealias Tuple = (measurement: String,
                          tags: [String?: String?]?,
                          fields: [String?: InfluxDBClient.Point.FieldValue?],
                          time: InfluxDBClient.Point.TimestampValue?)
```